cmd/driver-registrar/main.go

```diff
@@ -51,11 +41,33 @@ const (
 
 	// Verify (and update, if needed) the node ID at this freqeuency.
 	sleepDuration = 2 * time.Minute
+
+	modeNodeRegister = "node-register"
+	modeKubeRegister = "kubernetes-register"
+	defaultMode      = modeNodeRegister
 )
 
 // Command line flags
 var (
-	kubeconfig              = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
+	kubeconfig = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
+	runMode    = flag.String("mode", defaultMode, "Mode to run the program. Supported modes: node-register, kubernetes-register.\n"+
+		"In "+modeNodeRegister+" mode, the program will register the CSI driver with the "+
+		"node, setting up the node information accordingly.\n"+
+		"In "+modeKubeRegister+" mode, the program will register the driver with "+
```

Comment:
Seems that in both modes would require a replicaset of 1.  In NodeRegister, the CSIDriver should be registered once as well as registration with Kubernetes.  Also, from a usability point of view, it's not clear why user/admin would select one mode over the other (I guess this will need documenting).

Reply:
Yeah, that's for the docs to provide